pyannote/audio/labeling/tasks/multilabel.py

```diff
@@ -0,0 +1,306 @@
+        Each generator will prefetch enough batches to cover a whole epoch.
+        Set `parallel` to 0 to not use background generators.
+
+    Usage
+    -----
+    # precomputed features
+    >>> from pyannote.audio.features import Precomputed
+    >>> precomputed = Precomputed('/path/to/mfcc')
+
+    # instantiate batch generator
+    >>> batches =  MultilabelGenerator(precomputed)
+
+    # evaluation protocol
+    >>> from pyannote.database import get_protocol
+    >>> protocol = get_protocol('BabyTrain.SpeakerRole.JSALT')
+
+    # iterate over training set
+    >>> for batch in batches(protocol, subset='train'):
+    >>>     # batch['X'] is a (batch_size, n_samples, n_features) numpy array
+    >>>     # batch['y'] is a (batch_size, n_samples, n_tasks) numpy array
+    >>>     pass
+    """
+
+    def __init__(self, feature_extraction, protocol, labels,
+                 subset='train', frame_info=None, frame_crop=None,
+                 duration=3.2, batch_size=32, per_epoch=1, parallel=1,
+                 shuffle=True):
+
+        self.labels_spec = labels
+        super().__init__(feature_extraction, protocol, subset=subset,
+                         frame_info=frame_info, frame_crop=frame_crop,
+                         duration=duration,
+                         batch_size=batch_size, per_epoch=per_epoch,
+                         parallel=parallel, shuffle=shuffle)
+
+    def initialize_y(self, current_file):
+        # First, one hot encode the regular classes
+        annotation = current_file['annotation'].subset(self.labels_spec['regular'])
+        y, _ = one_hot_encoding(annotation,
+                                get_annotated(current_file),
+                                self.frame_info,
+                                labels=self.labels_spec["regular"],
+                                mode='center')
+        y_data = y.data
+        # Then, one hot encode the meta classes
+        for derivation_type in ['union', 'intersection']:
+            for meta_label, regular_labels in self.labels_spec[derivation_type].items():
+                derived = derives_label(current_file["annotation"], derivation_type, meta_label, regular_labels)
+                z, _ = one_hot_encoding(derived, get_annotated(current_file),
+                                        self.frame_info,
+                                        labels=[meta_label],
+                                        mode='center')
+
+                y_data = np.hstack((y_data, z.data))
+
+        return SlidingWindowFeature(self.postprocess_y(y_data),
+                                    y.sliding_window)
+
+    @property
+    def specifications(self):
+        return {
+            'task': TASK_MULTI_LABEL_CLASSIFICATION,
+            'X': {'dimension': self.feature_extraction.dimension},
+            'y': {'classes': self.labels_spec["regular"]
+                             + list(self.labels_spec['union'].keys())
```

Comment:
```suggestion
                             + list(self.labels_spec['union'])
```

No need for `.keys()`